src/Datadog.Trace.DuckTyping/DuckType.cs

```diff
@@ -0,0 +1,499 @@
+                            // Check if the target property can be written
+                            if (!targetProperty.CanWrite)
+                            {
+                                throw new DuckTypePropertyCantBeWrittenException(targetProperty);
+                            }
+
+                            // Check if the target property declaring type is an struct (structs modification is not supported)
+                            if (targetProperty.DeclaringType.IsValueType)
+                            {
+                                throw new DuckTypeStructMembersCannotBeChangedException(targetProperty.DeclaringType);
+                            }
+
+                            propertyBuilder.SetSetMethod(GetPropertySetMethod(proxyTypeBuilder, targetType, proxyProperty, targetProperty, instanceField));
+                        }
+
+                        break;
+
+                    case DuckKind.Field:
+                        FieldInfo targetField = targetType.GetField(duckAttribute.Name, DefaultFlags);
+                        if (targetField is null)
+                        {
+                            break;
+                        }
+
+                        propertyBuilder ??= proxyTypeBuilder.DefineProperty(proxyProperty.Name, PropertyAttributes.None, proxyProperty.PropertyType, null);
+
+                        if (proxyProperty.CanRead)
+                        {
+                            propertyBuilder.SetGetMethod(GetFieldGetMethod(proxyTypeBuilder, targetType, proxyProperty, targetField, instanceField));
+                        }
+
+                        if (proxyProperty.CanWrite)
+                        {
+                            // Check if the target field is marked as InitOnly (readonly) and throw an exception in that case
+                            if ((targetField.Attributes & FieldAttributes.InitOnly) != 0)
+                            {
+                                throw new DuckTypeFieldIsReadonlyException(targetField);
+                            }
+
+                            // Check if the target field declaring type is an struct (structs modification is not supported)
+                            if (targetField.DeclaringType.IsValueType)
+                            {
+                                throw new DuckTypeStructMembersCannotBeChangedException(targetField.DeclaringType);
+                            }
+
+                            propertyBuilder.SetSetMethod(GetFieldSetMethod(proxyTypeBuilder, targetType, proxyProperty, targetField, instanceField));
+                        }
+
+                        break;
+                }
+
+                if (propertyBuilder is null)
+                {
+                    continue;
+                }
+
+                if (proxyProperty.CanRead && propertyBuilder.GetMethod is null)
+                {
+                    throw new DuckTypePropertyOrFieldNotFoundException(proxyProperty.Name);
```

Comment:
Suggestion, could we include both the proxyProperty.Name and the duckAttribute.Name in the exception?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/946/commits/e3020964dcbb2b80b94368777abbdd5b1de4e248